judge/judgedaemon.main.php

```diff
@@ -16,6 +16,32 @@
 $endpoints = [];
 $domjudge_config = [];
 
+function dj_getopt(string $short_options, array $long_options = []): array
+{
+    global $argv;
+    define('GETOPT_REGEX', "/^([a-zA-Z0-9]:{0,2})*$/");
+    if (preg_match(GETOPT_REGEX, $short_options) !== 1) {
+        echo "Error: short options format specified is invalid.\n";
+        usage();
+    }
+    $options = getopt($short_options, $long_options);
+    if ($options===false || !is_array($argv)) {
+        echo "Error: parsing options failed.\nPlease check: `register_argc_arg` in php.ini.\n";
+        usage();
+    }
+    $unknown = false;
```

Comment:
I think this would be clearer:
```suggestion
    $unknown_option = false;
```